src/SmartComponents/RunTaskModal/ParameterTextbox.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import propTypes from 'prop-types';
+import {
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  TextInput,
+} from '@patternfly/react-core';
+
+const ParameterTextbox = ({
+  parameter,
+  updateParameters,
+  customDescription = undefined,
+}) => {
+  const [parameterText, setParameterText] = useState(
+    parameter.default || parameter.value
+  );
+  const [validated, setValidated] = useState('default');
+  const [helperText, setHelperText] = useState(
+    customDescription || parameter.description
+  );
+
+  return (
+    <Form className="pf-v5-u-pb-lg">
+      <FormGroup
+        label={parameter.title || parameter.key}
+        isRequired={parameter.required}
+        type="text"
+        fieldId="name"
+      >
+        <TextInput
+          value={parameterText}
+          type="text"
+          onChange={(_event, text) => {
+            if (parameter.required) {
+              if (text.trim().length) {
+                setValidated('success');
+                setHelperText(parameter.description);
+                setParameterText(text);
+                updateParameters(text);
```

Comment:
IIRC, the function takes two arguments:
```suggestion
                updateParameters(parameter.key, text);
```

Reply:
Done!